libs/phpseclib/vendor/phpseclib/phpseclib/phpseclib/Crypt/Hash.php

```
@@ -1445,6 +1736,151 @@ private static function sha512($m, $hash)
+                self::add64($hash[3], $d),
+                self::add64($hash[4], $e),
+                self::add64($hash[5], $f),
+                self::add64($hash[6], $g),
+                self::add64($hash[7], $h),
+            ];
+        }
+
+        // Produce the final hash value (big-endian)
+        // (\phpseclib3\Crypt\Hash::hash() trims the output for hashes but not for HMACs.  as such, we trim the output here)
+        return pack('J*', ...$hash);
+    }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'sha512_64'. (Unused Code Rules)

(UnusedPrivateMethod)